install_config/advanced_ldap_configuration/sssd_for_ldap_failover.adoc

```diff
@@ -0,0 +1,425 @@
+  RewriteCond %{REQUEST_URI} ^/challenging-proxy
+  RewriteCond %{HTTP:X-Csrf-Token} ^$ [NC]
+  RewriteRule ^.* - [F,L]
+
+  <Location /challenging-proxy/oauth/authorize>
+    # Insert your backend server name/ip here.
+    ProxyPass https://openshift.example.com:8443/oauth/authorize
+    AuthType Basic
+    AuthBasicProvider PAM
+    AuthPAMService openshift
+    Require valid-user
+  </Location>
+
+  <ProxyMatch /oauth/authorize>
+    AuthName openshift
+    RequestHeader set X-Remote-User %{REMOTE_USER}s
+  </ProxyMatch>
+</VirtualHost>
+
+RequestHeader unset X-Remote-User
+----
+====
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how to add the *login-proxy* block to
+support form authentication.
+====
+
+. Set a boolean to tell SELinux that it is acceptable for Apache to contact the
+PAM subsystem:
++
+====
+----
+# setsebool -P allow_httpd_mod_auth_pam on
+----
+====
+
+. Start up Apache:
++
+====
+----
+# systemctl start httpd.service
+----
+====
+
+[[sssd-phase-3-openshift-configuration]]
+== Phase 3: {product-title} Configuration
+
+This section describes how to set up an {product-title} server from scratch in
+an "all in one" configuration.
+link:../master_node_configuration.html[Master and Node
+Configuration] provides more information on alternate configurations.
+
+Modify the default configuration to use the new identity provider just
+created. To do so:
+
+. Modify the *_/etc/origin/master/master-config.yaml_* file.
+
+. Scan through it and locate the *identityProviders* section and replace it with:
++
+====
+----
+  identityProviders:
+  - name: any_provider_name
+    challenge: true
+    login: false
+    mappingMethod: claim
+    provider:
+      apiVersion: v1
+      kind: RequestHeaderIdentityProvider
+      challengeURL: "https://proxy.example.com/challenging-proxy/oauth/authorize?${query}"
+      clientCA: /etc/origin/master/proxy/proxyca.crt
+      headers:
+      - X-Remote-User
+----
+====
++
+[NOTE]
+====
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] explains how to add the login URL to support web
+logins.
+
+link:../advanced_ldap_configuration/configuring_extended_ldap_attributes.html[Configuring
+Extended LDAP Attributes] explains how to add the email and full-name
+attributes.
+====
+
+. Start {product-title} with the updated configuration:
++
+====
+----
+# openshift start \
+    --public-master=https://openshift.example.com:8443 \
+    --master-config=/etc/origin/master/master-config.yaml \
+    --node-config=/etc/origin/node-node1.example.com/node-config.yaml
+----
+====
+
+. Test logins:
++
+====
+----
+oc login https://openshift.example.com:8443
+----
+====
++
+It should now be possible to log in with only valid LDAP credentials. Also, you
+should be able to test with the web login.
```

Comment:
At this point (assuming we don't merge the other pages), you would not yet be able to log in with the web login.